[CN-1113]: Improve scaling and upgrading doc (#150)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/scaling-upgrading.adoc

= Scaling and Upgrading a Hazelcast Deployment

To scale an existing Hazelcast deployment, modify the yaml definition you used to deploy Hazelcast cluster by updating `clusterSize` field and re-apply it again.

Assuming the following is the existing configuration:

[source,yaml]
----

> apiVersion: hazelcast.com/v1alpha1

Reviewer: Could you add `kubectl scale` option as well? It will be available with 5.11 release.

Author: https://github.com/hazelcast/hazelcast-platform-operator-docs/pull/150/commits/8854a2f819706699a1324f037a487a540fe70da3

= Scaling and Upgrading a Hazelcast Deployment

To scale an existing Hazelcast deployment, there are two options:

> 

Reviewer 2: ```suggestion
+
```

Reviewer 2: Add this plus on this line and on each empty line related to the list items.

This is because adding a plus on empty lines that are related to a list item indents the content to make it clearer where the two options are (the list items) and what relates to that option.

- Modify the yaml definition you used to deploy Hazelcast cluster by updating `clusterSize` field and re-apply it again:

Assuming the following is the existing configuration:

[source,yaml]
----
apiVersion: hazelcast.com/v1alpha1
kind: Hazelcast
metadata:
  name: hazelcast-sample
spec:
  clusterSize: 3
----

> and you want to scale your deployment to 5 replicas, you need to apply the following:

Reviewer 2: ```suggestion
and you want to scale your deployment to five replicas, you need to apply the following:
```

Reviewer 2: rule of thumb with numbers (excluding measurements and that kind of thing) is one to ten in words, 11 and above in numbers.

== Upgrading

> Before upgrade, we recommend you to check link:https://docs.hazelcast.com/hazelcast/5.3/maintain-cluster/rolling-upgrades#hazelcast-members-compatibility-guarantees[Hazelcast's Compatibility Guarantees].

Reviewer 2: ```suggestion
Before upgrading, review the Hazelcast xref:maintain-cluster:rolling-upgrades.adoc#hazelcast-members-compatibility-guarantees[].
```

Author: Is it possible to reference to it like you did as `maintain-cluster:rolling-upgrades.adoc` is in another repository?: https://github.com/hazelcast/hz-docs

Reviewer 2: Sorry, I didn't pick up on that. I believe the following should work:

xref:{page-latest-supported-hazelcast}@hazelcast:maintain-cluster:rolling-upgrades.adoc#hazelcast-members-compatibility-guarantees[].

include::ROOT:example$/hazelcast.yaml[]
----

> There are two system properties worth mentioning related to upgrade:

Reviewer 2: ```suggestion
It is worth highlighting the following system properties, which are related to upgrades:
```

> - During rolling upgrade cluster is shutdown gracefully to prevent data loss. You might want to configure link:https://docs.hazelcast.com/hazelcast/5.3/system-properties#hazelcast.graceful.shutdown.max.wait[hazelcast.graceful.shutdown.max.wait system property].

Reviewer 2: ```suggestion
- During a rolling upgrade, the cluster is shutdown gracefully to prevent data loss. You can configure this using the link:https://docs.hazelcast.com/hazelcast/5.3/system-properties#hazelcast.graceful.shutdown.max.wait[hazelcast.graceful.shutdown.max.wait system property].
```

> - Last step of link:https://docs.hazelcast.com/hazelcast/5.3/maintain-cluster/rolling-upgrades#rolling-upgrade-procedure[rolling upgrade procedure] is to trigger a rolling upgrade on the cluster. Hazelcast Platform Operator triggers it automatically by setting `hazelcast.cluster.version.auto.upgrade.enabled` to `true` by default.

Reviewer 2: ```suggestion
- The final step in the link:https://docs.hazelcast.com/hazelcast/5.3/maintain-cluster/rolling-upgrades#rolling-upgrade-procedure[rolling upgrade procedure] is to trigger a rolling upgrade on the cluster. Hazelcast Platform Operator triggers this automatically by setting `hazelcast.cluster.version.auto.upgrade.enabled` to `true` by default.
```

> NOTE: You can check xref:jvm-parameters.adoc[JVM Parameters] documentation to learn how to configure.

Reviewer 2: ```suggestion
NOTE: For further information on the available configuration options, see the xref:configuration:jvm-parameters.adoc[] topic.
```

include::ROOT:example$/hazelcast-5.2.4.yaml[]
----

and you want to upgrade your Hazelcast version to {page-latest-supported-hazelcast}, you need to apply the following:

[source,yaml,subs="attributes+"]
----
include::ROOT:example$/hazelcast.yaml[]
----

It is worth highlighting the following system properties, which are related to upgrades:

- During a rolling upgrade, the cluster is shutdown gracefully to prevent data loss. You can configure this using the link:https://docs.hazelcast.com/hazelcast/5.3/system-properties#hazelcast.graceful.shutdown.max.wait[hazelcast.graceful.shutdown.max.wait system property].

- The final step in the link:https://docs.hazelcast.com/hazelcast/5.3/maintain-cluster/rolling-upgrades#rolling-upgrade-procedure[rolling upgrade procedure] is to trigger a rolling upgrade on the cluster. Hazelcast Platform Operator triggers it automatically by setting `hazelcast.cluster.version.auto.upgrade.enabled` to `true` by default, and it cannot be changed.

> - `hazelcast.persistence.auto.cluster.state` set to `true` by default and `hazelcast.persistence.auto.cluster.state.strategy` is set to `NO_MIGRATION` by default. They cannot be changed.

Reviewer: I think we need to move them to a separate section, as we discussed.

Author: Does it look good? https://github.com/hazelcast/hazelcast-platform-operator-docs/pull/150/commits/90200db7368650511cb4056b0d055324ca87198b

== Upgrading

> Before upgrade, we recommend you to check xref:hazelcast:maintain-cluster:rolling-upgrades.adoc#hazelcast-members-compatibility-guarantees[Rolling Upgrades].

Reviewer 2: ```suggestion
Before upgrading, we recommend that you read the xref:hazelcast:maintain-cluster:rolling-upgrades.adoc#hazelcast-members-compatibility-guarantees[Rolling Upgrades] information.
```
